cmds/core/sluinit/uinit.go

```diff
@@ -110,3 +113,68 @@ func unmountAndExit() {
 	time.Sleep(5 * time.Second) // let queued up debug statements get printed
 	os.Exit(1)
 }
+
+// scanIscsiDrives calls iscsinl to mount iscsi drives.
+// format: netroot=iscsi:@X.Y.Z.W::3260::iqn.FOO.com.abc:hostname-boot
+func scanIscsiDrives() error {
+
+	slaunch.Debug("Scanning kernel cmd line for *netroot* flag")
+	val, ok := cmdline.Flag("netroot")
+	if !ok {
+		return errors.New("netroot flag is not set")
+	}
+
+	// val = iscsi:@10.196.210.62::3260::iqn.1986-03.com.sun:ovs112-boot
+	slaunch.Debug("netroot flag is set with val=%s", val)
+	s := strings.Split(val, "::")
+	if len(s) != 3 {
+		return fmt.Errorf("%v: incorrect format ::,  Usage: netroot=iscsi:@10.X.Y.Z::1224::iqn.foo:hostname-bar, [Expecting len(%s) = 3] ", val, s)
+	}
+
+	// s[0] = iscsi:@10.196.210.62 or iscsi:@10.196.210.62,2
+	// s[1] = 3260
+	// s[2] = iqn.1986-03.com.sun:ovs112-boot
+	port := s[1]
+	volume := s[2]
+
+	// split s[0] into tmp[1] and tmp[2]
+	tmp := strings.Split(s[0], ":@")
```

Comment:
though i don't understand the @ in there, I didn't think that was part of the format?

Reply:
oh interesting, it allows username and password? 
       
```
netroot=iscsi:[<username>:<password>[:<reverse>:<password>]@][<servername>]:[<protocol>]:[<port>][:[<iscsi_iface_name>]:[<netdev_name>]]:[<LUN>]:<targetname>
```

nuts. could add support for that.

Reply:
and also supports `[:[<iscsi_iface_name>]:[<netdev_name>]]`

that's nuts. is that even parseable, given that the target name can have an arbitrary number of : in it?

Reply:
yeah, it's like impossible to do, so they rely on the target name to begin with iqn or eui or naa: https://github.com/dracutdevs/dracut/blob/7069132ec7e94a3274aa1b3b3b6e339066118a4d/modules.d/40network/net-lib.sh#L385

hate it when formats are not regular.

Reply:
yeah, i agree. the iscsi format is nuts. and in the field, I have seen all types of inputs parsed via kernel command line with expectation that iscsinl would parse it. So to make things simple (and limited) , I just allowed format of type 
val = iscsi:@10.196.210.62::3260::iqn.xyza-03.com.abc:ovs112-boot
i.e
val = iscsi:@IP::port::volume
which is limited but does the job for now....

I couldn't do what iscsistart does because of the different possible scenarios input can be passed. For example: the below is expected to be parsed and is a valid format...Note there is no '@' .
netroot=iscsi:10.196.210.62:::0:iqn.xyza-03.com.abc:ovs112-boot

Ideally, we should create something similar to the dracut script that you pointed out...and yes preferably  in iscsinl.

Reply:
how would you feel about adding user@pass support to the iscsi uri parser in pkg/dhclient?